Add some help to the command line

lontivero opened the pull request:
This PR add some help (not yet really) using the config metadata

```
$ wassabeed --help
Wasabi Daemon 2.0.4.0
Usage: Wasabi Daemon [OPTION]...

Available options are:
  --blockonlymode                  Indicate Wasabi should only listen for 
                                   blocks and not for transactions

  --coordinatoridentifier          -

  --downloadnewversion             Indicate new version of Wasabi should 
                                   be downloaded automatically

  --dustthreshold                  Threshold amount under which coin 
                                   received from others to reused 
                                   addresses are considered a dust attack

  --enablegpu                      Indicate Wasabi should use the GPU

  --jsonrpcpassword                The user password that is authorized to 
                                   make requests to the Json RPC server

  --jsonrpcserverenabled           Indicate the Json RPC Server should be 
                                   started and accepting requests

  --jsonrpcserverprefixes          Json RPC server prefixes

  --jsonrpcuser                    The user name that is authorized to 
                                   make requests to the Json RPC server

  --localbitcoincoredatadir        Specify the DataDir to be used by the 
                                   bitcoin node

  --loglevel                       Level of detail in the logs (trace, 
                                   debug, info, warning, error or 
                                   critical)

  --mainnetbackenduri              Url to the backend server to connect to 
                                   when network is main

  --mainnetbitcoinp2pendpoint      -

  --mainnetcoordinatoruri          Url to the coordinator server to 
                                   connect to when network is main

  --network                        Bitcoin network to use (main, testnet 
                                   or regtest)

  --regtestbackenduri              Url to the backend server to connect to 
                                   when network is regtest

  --regtestbitcoinp2pendpoint      -

  --regtestcoordinatoruri          Url to the coordinator server to 
                                   connect to when network is regtest

  --rpconionenabled                Indicate the Json RPC Server should be 
                                   published as an Tor Onion service

  --startlocalbitcoincoreonstartup Indicate that bitcoin node should be 
                                   started when wasabi starts

  --stoplocalbitcoincoreonshutdown Indicate that bitcoin node should be 
                                   stopped when finishes

  --terminatetoronexit             Indicate the Tor process must be 
                                   stopped when Wasabi finishes

  --testnetbackenduri              Url to the backend server to connect to 
                                   when network is testnet

  --testnetbitcoinp2pendpoint      -

  --testnetcoordinatoruri          Url to the coordinator server to 
                                   connect to when network is testnet

  --usetor                         Indicate all the communication have to 
                                   go thru the Tor network

MaxHillebrand reviewed (approved):
cACK, very helpful!

What does (source: config file) mean, and does it have to be mentioned in every line of the help output?

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Url to the coordinator server to connect to when network is testnet",
+				GetNullableStringValue("TestNetCoordinatorUri", PersistentConfig.TestNetCoordinatorUri, cliArgs)),
+			[ nameof(RegTestCoordinatorUri)] = (
+				"Url to the coordinator server to connect to when network is regtest",
```
```suggestion
				"The coordinator server's URL to connect to when the Bitcoin network is regtest",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Url to the coordinator server to connect to when network is regtest",
+				GetNullableStringValue("RegTestCoordinatorUri", PersistentConfig.RegTestCoordinatorUri, cliArgs)),
+			[ nameof(UseTor)] = (
+				"Indicate all the communication have to go thru the Tor network",
```
```suggestion
				"All the communications go through the Tor network",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Indicate all the communication have to go thru the Tor network",
+				GetBoolValue("UseTor", PersistentConfig.UseTor, cliArgs)),
+			[ nameof(TerminateTorOnExit)] = (
+				"Indicate the Tor process must be stopped when Wasabi finishes",
```
```suggestion
				"Stop the Tor process when Wasabi is closed",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Indicate the Tor process must be stopped when Wasabi finishes",
+				GetBoolValue("TerminateTorOnExit", PersistentConfig.TerminateTorOnExit, cliArgs)),
+			[ nameof(DownloadNewVersion)] = (
+				"Indicate new version of Wasabi should be downloaded automatically",
```
```suggestion
				"Automatically download any new released version of Wasabi",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Indicate new version of Wasabi should be downloaded automatically",
+				GetBoolValue("DownloadNewVersion", PersistentConfig.DownloadNewVersion, cliArgs)),
+			[ nameof(StartLocalBitcoinCoreOnStartup)] = (
+				"Indicate that bitcoin node should be started when wasabi starts",
```
```suggestion
				"Start a local bitcoin node when Wasabi starts",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Indicate that bitcoin node should be started when wasabi starts",
+				GetBoolValue("StartLocalBitcoinCoreOnStartup", PersistentConfig.StartLocalBitcoinCoreOnStartup, cliArgs)),
+			[ nameof(StopLocalBitcoinCoreOnShutdown)] = (
+				"Indicate that bitcoin node should be stopped when finishes",
```
```suggestion
				"Stop the local bitcoin node when Wasabi is closed",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Indicate that bitcoin node should be stopped when finishes",
+				GetBoolValue("StopLocalBitcoinCoreOnShutdown", PersistentConfig.StopLocalBitcoinCoreOnShutdown, cliArgs)),
+			[ nameof(LocalBitcoinCoreDataDir)] = (
+				"Specify the DataDir to be used by the bitcoin node",
```
```suggestion
				"The path of the data directory to be used by the local bitcoin node",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"-",
+				GetEndPointValue("RegTestBitcoinP2pEndPoint", PersistentConfig.RegTestBitcoinP2pEndPoint, cliArgs)),
+			[ nameof(JsonRpcServerEnabled)] = (
+				"Indicate the Json RPC Server should be started and accepting requests",
```
```suggestion
				"Start the Json RPC Server and accept requests",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"The user password that is authorized to make requests to the Json RPC server",
+				GetStringValue("JsonRpcPassword", PersistentConfig.JsonRpcPassword, cliArgs)),
+			[ nameof(JsonRpcServerPrefixes)] = (
+				"Json RPC server prefixes",
```
```suggestion
				"The Json RPC server prefixes",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Json RPC server prefixes",
+				GetStringArrayValue("JsonRpcServerPrefixes", PersistentConfig.JsonRpcServerPrefixes, cliArgs)),
+			[ nameof(RpcOnionEnabled)] = (
+				"Indicate the Json RPC Server should be published as an Tor Onion service",
```
```suggestion
				"Publish the Json RPC Server as a Tor Onion service",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Indicate the Json RPC Server should be published as an Tor Onion service",
+				GetBoolValue("RpcOnionEnabled", value: false, cliArgs)),
+			[ nameof(DustThreshold)] = (
+				"Threshold amount under which coin received from others to reused addresses are considered a dust attack",
```
```suggestion
				"The amount threshold under which coins received from others to reuse addresses are considered a dust attack",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Threshold amount under which coin received from others to reused addresses are considered a dust attack",
+				GetMoneyValue("DustThreshold", PersistentConfig.DustThreshold, cliArgs)),
+			[ nameof(BlockOnlyMode)] = (
+				"Indicate Wasabi should only listen for blocks and not for transactions",
```
```suggestion
				"Wasabi listens only for blocks and not for transactions",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Indicate Wasabi should only listen for blocks and not for transactions",
+				GetBoolValue("BlockOnly", value: false, cliArgs)),
+			[ nameof(LogLevel)] = (
+				"Level of detail in the logs (trace, debug, info, warning, error or critical)",
```
```suggestion
				"The level of detail in the logs: trace, debug, info, warning, error, or critical",
```

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Level of detail in the logs (trace, debug, info, warning, error or critical)",
+				GetStringValue("LogLevel", value: "", cliArgs)),
+			[ nameof(EnableGpu)] = (
+				"Indicate Wasabi should use the GPU",
```
```suggestion
				"Use a GPU to render the user interface",
```

yahiheb reviewed (changes requested):
Small tweaks to the help text.

yahiheb commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Publish the Json RPC Server as a Tor Onion service",
+				GetBoolValue("RpcOnionEnabled", value: false, cliArgs)),
+			[ nameof(DustThreshold)] = (
+				"The amount threshold under which coins received from others to reuse addresses are considered a dust attack",
```
```suggestion
				"The amount threshold under which coins received from others to already used addresses are considered a dust attack",
```

yahiheb commented:
> @yahiheb the location is URL because is to identify a resource in the internet. URI is a bit more generic.

My bad, I changed that back to URL.

lontivero commented:
Ok, let's go with URI then, it's not wrong. I added the default values in the first version but it was difficult to explain where those values came from, that's why i removed them.

MaxHillebrand commented on WalletWasabi.Daemon/Config.cs:
```diff
@@ -21,36 +22,88 @@ public Config(PersistentConfig persistentConfig, string[] cliArgs)
+				"Stop the Tor process when Wasabi is closed",
+				GetBoolValue("TerminateTorOnExit", PersistentConfig.TerminateTorOnExit, cliArgs)),
+			[ nameof(DownloadNewVersion)] = (
+				"Automatically download any new released version of Wasabi",
```
Is it any new version, or just the major version?

MarnixCroes replied:
any new version.
In the GUI the _update available_ message is only with major version.
Auto download always gets the latest release
